multiscale_run/neurodamus_manager.py

```diff
@@ -428,3 +430,59 @@ def gen_radii(astro_id):
         assert len(vasc_ids) == len(radii), f"{len(vasc_ids)} == {len(radii)}"
 
         return vasc_ids, radii
+
+    @staticmethod
+    def stats():
+        """Extract general statistics from a simulation
+
+        simulation_config.json is expected to be in the folder.
+        """
+
+        simulation_config = libsonata.SimulationConfig.from_file(
+            "simulation_config.json"
+        )
+        circuit_config = libsonata.CircuitConfig.from_file(simulation_config.network)
+        node_sets = libsonata.NodeSets.from_file(circuit_config.node_sets_path)
+        node_sets.update(libsonata.NodeSets.from_file(simulation_config.node_sets_file))
+
+        def print_attribute(attribute, pop, selection):
+            """Print attribute if present"""
+            if attribute in pop.attribute_names:
+                with np.printoptions(threshold=3):
+                    print(f"{attribute}: (frequency) gids")
+                    attr = pop.get_attribute(attribute, selection)
+                    d = {}
+                    for idx, val in zip(selection.flatten(), attr):
+                        if val not in d:
+                            d[val] = []
+                        d[val].append(idx)
+                    d = sorted(d.items())
+                    print(
+                        "    "
+                        + "\n    ".join(
+                            f"{k[0]}: ({len(k[1])}) {np.array(k[1])}" for k in d
+                        )
+                    )
```

Comment:
This would seem like a good use case for https://docs.python.org/3/library/textwrap.html#textwrap.indent

Reply:
done